vm/src/obj/objfloat.rs

```diff
@@ -213,8 +213,11 @@ impl PyFloat {
 
     #[pymethod(name = "__format__")]
     fn format(&self, spec: PyStringRef, vm: &VirtualMachine) -> PyResult<String> {
-        let format_spec = FormatSpec::parse(spec.as_str());
-        match format_spec.format_float(self.value) {
+        let try_format = || {
+            let format_spec = FormatSpec::parse(spec.as_str())?;
+            format_spec.format_float(self.value)
+        };
+        match try_format() {
```

Comment:
You could do something like
```rust
match FormatSpec::parse(spec.as_str()).and_then(|spec| spec.format_float(self.value)) {
    ...
}
```
to avoid the closure.

Reply:
Sure thing. Out of curiosity, what are the benefits (if any) of doing as shown above vs. using a closure? Or is it just Rust convention?

Reply:
Since it's not a very complicated operation, it's simpler to use the combinator method than define a closure and immediately call it once. It's similar to `do` vs `>>=` if you've worked with Haskell ([do notation considered harmful](https://wiki.haskell.org/Do_notation_considered_harmful)).